rpc/src/test/scala/org/bitcoins/rpc/RpcClientTest.scala

```diff
@@ -0,0 +1,517 @@
+        assert(info.size == 0)
+        sendTransaction.flatMap { _ =>
+          client.getMemPoolInfo.map { newInfo =>
+            assert(newInfo.size == 1)
+          }
+        }
+      }
+    }
+  }
+
+  it should "be able to get a transaction" in { // Assumes a block
+    client.getBlockHash(1).flatMap { blockHash =>
+      client.getBlock(blockHash).flatMap { block =>
+        client.getTransaction(block.tx.head).flatMap { tx =>
+          assert(tx.txid == block.tx.head)
+          assert(tx.amount == Bitcoins(50))
+          assert(tx.blockindex.get == 0)
+          assert(tx.details.head.category == "generate")
+          assert(tx.generated.get)
+          client.getBlockCount.map { count =>
+            assert(tx.confirmations == count)
+          }
+        }
+      }
+    }
+  }
+
+  it should "be able to get the network hash per sec" in {
+    client.getNetworkHashPS().map { hps =>
+      assert(hps > 0)
+    }
+  }
+
+  it should "be able to get an address from bitcoind" in {
+    val addressF = client.getNewAddress()
+    addressF.map { address =>
+      logger.info(address.value)
+      assert(true)
+    }
+  }
+
+  it should "be able to get a new raw change address" in {
+    client.getRawChangeAddress().map { address =>
+      assert(true)
+    }
+
+    client.getRawChangeAddress(Some("legacy")).map { address =>
+      assert(true)
+    }
+
+    client.getRawChangeAddress(Some("p2sh-segwit")).map { address =>
+      assert(true)
+    }
+
+    client.getRawChangeAddress(Some("bech32")).map { address =>
+      assert(true)
+    }
+  }
+
+  it should "be able to get the amount recieved by some address" in {
+    client.getNewAddress().flatMap { address =>
+      client.getReceivedByAddress(address).flatMap { amount =>
+        assert(amount == Bitcoins(0))
+      }
+    }
+  }
+
+  it should "be able to get the tx outset info" in {
+    client.getTxOutSetInfo.flatMap { info =>
+      client.getBlockCount.map { count =>
+        assert(info.height == count)
+      }
+      client.getBestBlockHash.map { hash =>
+        assert(info.bestblock == hash)
+      }
+    }
+  }
+
+  it should "be able to get the unconfirmed balance" in {
+    client.getUnconfirmedBalance.flatMap { balance =>
+      assert(balance == Bitcoins(0))
+      sendTransaction.flatMap { transaction =>
+        client.getUnconfirmedBalance.map { newBalance =>
+          assert(newBalance == transaction.amount)
+        }
+      }
+    }
+  }
+
+  it should "be able to refill the keypool" in {
+    client.getWalletInfo.flatMap { info =>
+      client.keyPoolRefill(info.keypoolsize + 1).flatMap { _ =>
+        client.getWalletInfo.flatMap { newInfo =>
+          assert(newInfo.keypoolsize == info.keypoolsize + 1)
+        }
+      }
+    }
+  }
+
+  it should "be able to get the block count" in {
+    val blockCountF = client.getBlockCount
+    blockCountF.map { count =>
+      logger.info(count.toString)
+      assert(count >= 0)
```

Comment:
`assert(count > 0)` ? We should have a block at this point right? Especially if we end up setting up blocks in a `beforeAll`